### PR TITLE
[ready for review] smc issue #825 sage_eval in sagews

### PR DESCRIPTION
**Pull request by DrXyzzy**
Ref: #825 

Test file https://cloud.sagemath.com/projects/ccd4d4a4-29a8-4c39-85c2-a630cb1e9b6c/files/SAGEWS/issue825_test.sagews
Project is on smc branch for test code at this time.

### CODE REVIEW

File: src/smc_sagews/smc_sagews/sage_salvus.py [hunk 1]
```diff
@@ -616,12 +616,13 @@ def interact_control(arg, value):
...
     return c
 
-def sage_eval(x, locals=None):
-    x = str(x).strip()
-    if x.isspace():
-        return None
+def sage_eval(x, locals=None, **kwds):
+    if isinstance(x, str):
+        x = str(x).strip()
+        if x.isspace():
```
Comment (haraldschilly) [inline src/smc_sagews/smc_sagews/sage_salvus.py]: I'm sorry, I don't get this test. If it is a space, isn't it then the empty string?

`" ".strip().isspace()` → `False`

**Comment by haraldschilly** [comment]
added one question, maybe I don't get it, otherwise ok. the main idea is to pass on the kwds and that's done

**Comment by DrXyzzy** [comment]
@haraldschilly,
Thank you for reviewing.
- The bug initially reported in #825 appears to be fixed by @williamstein suggestion to insert `if isinstance(x, str):`.
- The bug found during testing about ignoring preparse named parameter appears to be fixed, as you noted, by passing through kwds.
- The isspace code was there before. I didn't change it. I wonder if it relates to this: [str.strip() strips ASCII whitespace while isspace() is locale-dependent](https://bugs.python.org/issue25433)
- agree with suggestion about need for testing, have started on that.

**Comment by haraldschilly** [comment]
oh ok, I see. ... and that locale dependency is good to know, could also happen in other places.

**Pull request merged**